backend/substrapp/tasks/utils.py

```diff
@@ -88,15 +86,23 @@ def get_memory_limit():
         'cap_drop': ['ALL'],
     }
 
+    memory_limit_bytes = docker_client.containers.run(**task_args)
+
+    return int(memory_limit_bytes)
+
+
+@timeit
+def get_memory_limit():
+
     memory_limit = local_memory_limit()
 
     try:
-        memory_limit_bytes = docker_client.containers.run(**task_args)
-    except (docker.errors.ContainerError, docker.errors.ImageNotFound, docker.errors.APIError):
-        logger.info('[Warning] Cannot get memory limit from remote')
-    else:
-        if memory_limit_bytes:
-            memory_limit = int(memory_limit_bytes)
+        memory_limit = docker_memory_limit()
+    except (docker.errors.ContainerError, docker.errors.ImageNotFound,
+            docker.errors.APIError, ValueError):
+
+        logger.info('[Warning] Cannot get memory limit from remote, get it from local.')
+        memory_limit = local_memory_limit()
```

Comment:
we're calling local_memory_limit twice (also on line 97). refactor?

Reply:
Yes !

Reply:
@Kelvin-M what about other exceptions that aren't covered on line 101-102?

Reply:
It should not raise, if so we let the worker raise the exception :)

Reply:
I don't understand what's the advantage of only catching specific exception types. 

If the wrapped method raises a new exception type in the future (e.g. following code changes in the wrapped method, or changes in a library), then the wrapper method will return nothing 😱 

I'll leave that up to you though

Reply:
Here, we just want to catch exceptions that could happen and we are aware of.
In other case, I prefer to let the function raise an exception.
I don't understand the "wrapped method" part. For me it's just a classical function (with a decorator)

Reply:
That makes sense, I withdraw my comment! (I had mistakenly assumed uncaught exceptions will not be propagated)

Re: wrapper/wrapped. The wrapped functions are the two functions that actually get the memory limit (docker/memory). The wrapper function is `get_memory_limit`. I used "wrapping" here to refer to nesting (one function calls another one, Russian-dolls like)